rest/server_context.go

```diff
@@ -447,6 +447,16 @@ func (sc *ServerContext) _getOrAddDatabaseFromConfig(ctx context.Context, config
 		if !bucket.IsSupported(sgbucket.DataStoreFeatureCollections) {
 			return nil, errCollectionsUnsupported
 		}
+		// TODO: This needs changing when implementing Phase 2 collections: support for multiple collections
+		collection, err := base.AsCollection(bucket)
+		if err != nil {
+			return nil, err
+		}
+		// Check if scope/collection specified exists
+		_, err = collection.Exists("waitUntilScopeAndCollectionExists", nil)
```

Comment:
I wonder whether we want to apply the same retry logic to non-existing collections as we do for non-existing buckets, to allow scripted startup/collection creation.

See inside `connectToBucket` for a reference as to what we do regarding retry when opening buckets.

It may be that we make the existing util function a non-test one so we can use it here.